specreduce/tests/test_line_matching.py

```diff
@@ -0,0 +1,110 @@
+import pytest
+
+import numpy as np
+
+from astropy.wcs import WCS
+from astropy.modeling import models
+from astropy.nddata import StdDevUncertainty
+
+from specutils import Spectrum1D
+from specutils.fitting import fit_generic_continuum
+
+from specreduce.extract import BoxcarExtract
+from specreduce.utils.synth_data import make_2d_arc_image
+from specreduce.tracing import FlatTrace
+from specreduce.calibration_data import load_pypeit_calibration_lines
+from specreduce.line_matching import match_lines_wcs, find_arc_lines
+
+
+@pytest.mark.remote_data
+@pytest.fixture
+def mk_test_data():
+    """
+    Create test data for the line matching routines.
+    """
+    non_linear_header = {
+        'CTYPE1': 'AWAV-GRA',  # Grating dispersion function with air wavelengths
+        'CUNIT1': 'Angstrom',  # Dispersion units
+        'CRPIX1': 519.8,       # Reference pixel [pix]
+        'CRVAL1': 7245.2,      # Reference value [Angstrom]
+        'CDELT1': 2.956,       # Linear dispersion [Angstrom/pix]
+        'PV1_0': 4.5e5,        # Grating density [1/m]
+        'PV1_1': 1,            # Diffraction order
+        'PV1_2': 27.0,         # Incident angle [deg]
+        'PV1_3': 1.765,        # Reference refraction
+        'PV1_4': -1.077e6,     # Refraction derivative [1/m]
+        'CTYPE2': 'PIXEL',     # Spatial detector coordinates
+        'CUNIT2': 'pix',       # Spatial units
+        'CRPIX2': 1,           # Reference pixel
+        'CRVAL2': 0,           # Reference value
+        'CDELT2': 1            # Spatial units per pixel
+    }
+    linear_header = {
+        'CTYPE1': 'AWAV',  # Grating dispersion function with air wavelengths
+        'CUNIT1': 'Angstrom',  # Dispersion units
+        'CRPIX1': 519.8,       # Reference pixel [pix]
+        'CRVAL1': 7245.2,      # Reference value [Angstrom]
+        'CDELT1': 2.956,       # Linear dispersion [Angstrom/pix]
+        'CTYPE2': 'PIXEL',     # Spatial detector coordinates
+        'CUNIT2': 'pix',       # Spatial units
+        'CRPIX2': 1,           # Reference pixel
+        'CRVAL2': 0,           # Reference value
+        'CDELT2': 1            # Spatial units per pixel
+    }
+    non_linear_wcs = WCS(header=non_linear_header)
```

Comment:
does it make sense to also test this with GWCS since the function can accept either? im not sure if thats overkill